src/util/threadpool.rs

```diff
@@ -173,36 +167,53 @@ where
     }
 }
 
+pub trait Context: Send {
+    fn on_task_started(&mut self, worker_ctx: &mut Self);
```

Comment:
Do something about it. Checking or adding or something. This is the only way to change the worker context.